experimental/storage/layered-map/src/lib.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    dropper::DROPPER,
+    metrics::{LAYER, TIMER},
+    node::{NodeRef, NodeStrongRef},
+};
+use aptos_crypto::HashValue;
+use aptos_infallible::Mutex;
+use aptos_metrics_core::{IntGaugeHelper, TimerHelper};
+use std::sync::Arc;
+
+mod dropper;
+mod metrics;
+mod node;
+pub(crate) mod r#ref;
+
+#[cfg(test)]
+mod tests;
+
+/// When recursively creating a new `MapLayer` (a crit bit tree overlay), passing down `Vec<(K, Option<V>)>`
+/// That's why we require `Key: Clone` and clone the key and value only when the leaf node is
+/// created.
+pub trait Key: Clone + Send + Sync + Eq + 'static {
+    fn iter_bits(&self) -> impl Iterator<Item = bool>;
+
+    fn bit(&self, depth: usize) -> bool;
+}
+
+/// Similar to `Key`, we require `Value: Clone`, another reason being it's tricky to figure out the
+/// lifetime if `get()` returns a reference to the value -- we simply clone the value.
+pub trait Value: Clone + Send + Sync + 'static {}
+
+#[derive(Debug)]
+struct LayerInner<K: Key, V: Value> {
+    root: NodeRef<K, V>,
+    children: Mutex<Vec<Arc<LayerInner<K, V>>>>,
```

Comment:
what is children used for?

Reply:
same with SMT, there can be branches in the block tree.